config/config.exs

```diff
@@ -1,4 +1,6 @@
 import Config
 
-config :cim,
-  cowboy_port: 8080
+if System.get_env("COWBOY_PORT") do
+  config :cim,
+    cowboy_port: System.get_env("COWBOY_PORT") |> String.to_integer()
+end
```

Comment:
this one needs to go in config/runtime.exs.  config/config.exs is read at compile-time, runtime.exs is read at runtime.  It will work fine in development/test having them in config as your app is recompiled on every change, but it won't work in production because whatever that variable was when the app was compiled for release will be stuck there forever :)